### ISSUE TITLE
ND is showing strange and wrong route

### ISSUE THREAD

**Issue by flytester** [flybywiresim/a32nx#6645]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2022-02-03T09:40:14+00:00",
  "ref": "master",
  "sha": "a5566f397b74f77ba33edc9f1ad7f4da8d303ce4",
  "actor": "aguther",
  "event_name": "manual"
}
```

### Describe the bug

When I import a correct flight plan from simbrief, the route displayed by the ND is very often a complete mesh with lines to non-existing waypoints (see picture / flightplan is EGGP-EDDW). This has been an issue for quit a while now. Occurs only in the dev/exp version, stable version doesn't seem to have this bug. Sometimes you are lucky and all works fine, but very often a strange and obviously wrong flight pad shows up.
![FlightSimulator_2022 02 03-20 34](https://user-images.githubusercontent.com/73710754/152416186-5b7d17a3-0b8f-42fa-9aa9-77542991317c.jpg)

### Expected behavior

ND shows flight plan, including SID/STAR, correctly.

### Steps to reproduce

-Import SIM-brief flightplan
-Add / Alter desired runways and SID/STAR
-Check Flight plan on ND in plan-mode

**Comment by Benjozork** [comment]
Need your routing. Or screenshots of the complete F-PLN page.

**Comment by kiokio0987** [comment]
I checked to see if this was really happening.
As a result, in regards to the image,Your flight plan may have been set incorrectly.
However, I found out that there is one thing that is not supported by A32NX.

if STAR is set to "No STAR", "VIA" cannot be selected.

development & experimental

Build info (development)
{
  "built": "2022-02-03T09:40:14+00:00",
  "ref": "master",
  "sha": "a5566f397b74f77ba33edc9f1ad7f4da8d303ce4",
  "actor": "aguther",
  "event_name": "manual"
}

Build info (experimental)
{
  "built": "2022-02-03T09:46:08+00:00",
  "ref": "experimental",
  "sha": "cbc23c36a68eb26d90d7fed02b466900e269f374",
  "actor": "aguther",
  "event_name": "manual"
}

Flight Plans
EGGP/27 BART1T BARTN UL975 DESIG L603 LAMSO DCT PETIK DCT ANDIK DCT EEL N125 AVESA DCT GIBMA GIBM2P EDDW/GIBMA.ILS 27-Z

@flytester It looks like your route uses STAR, "GIBMA2P".
![GIBMA2P-1-1](https://user-images.githubusercontent.com/75520968/152470359-51b977ea-5b1b-4bec-b4d0-268032f3873f.jpg)

Probably on the ARRIVAL page, you probably entered it like this?
This way I was able to reproduce the problem as shown in the image.
![dev_MCDU2](https://user-images.githubusercontent.com/75520968/152477486-c81740e9-f3c9-4485-843a-5596a287fdcd.jpg)

I checked navigraph and it seems that the "GIBMA" part of "VIA" needs to be changed to "BMN".
![dev_MCDU3](https://user-images.githubusercontent.com/75520968/152477611-c09443e9-d5e5-4365-92ab-382d2e008c22.jpg)
So the route shown in ND will look like this.(When using STAR's "GIBMA2P")
![dev_gibma2p_3](https://user-images.githubusercontent.com/75520968/152467401-1c7fb9b8-db65-43d2-a45b-8d3c3120bb83.jpg)

Here are the commands that A32NX may not support.
If you take the route with a large right turn and pass through "DW555", there is no need for STAR. (GIB27)

GIB27(EDDW)
![Navigraph Charts-1](https://user-images.githubusercontent.com/75520968/152470537-bbb09495-ba1e-48ed-a2c6-54b13abbc071.jpg)

However, vias are required. (Confirmed with Toliss A321. I'm sorry, but I'm not a pilot, so I don't know if this is a real operation.)
Appr:ILS 27 Z, via:GIB27,　STAR:NONE 
This is how Toliss A321 describes it.
In the case of A32NX, if STAR is set to "No STAR", "VIA" cannot be selected.

 Toliss A321 (Appr: ILS 27 Z, via: GIB27, STAR: NONE)
![xplane-1](https://user-images.githubusercontent.com/75520968/152477374-a6e817f6-3383-4820-9f7b-c3c505745e8d.jpg)


This is happening for both "development" and "Expremental".
The only way to do this is to check the chart and add the WAYPOINT later.
However, I don't think this is a bug, as I think this is a feature that hasn't been added yet.
![dev](https://user-images.githubusercontent.com/75520968/152468702-7f549f8a-8481-4ae3-b820-5ab02a489c61.jpg)

**Comment by flytester** [comment]
EGGP/27 NANT2T NANTI Y53 MOGTA UY124 LISTO UL612 HON P155 SIVDA P137 REDFA L620 TULIP Y12 SPY N873 GRONY M105 EEL N125 AVESA T560 GIBMA GIBM2P EDDW/GIBMA.I27-Z
[EGGPEDDW(rename to pln).txt](https://github.com/flybywiresim/a32nx/files/8005119/EGGPEDDW.rename.to.pln.txt)

**Comment by 0x070E** [comment]
Yes, GIBMA should connect to DW402 to enter the right downwind to RWY 27

**Comment by derl30n** [comment]
bump

**Comment by tracernz** [comment]
Testing on master ([3528d3e](https://github.com/flybywiresim/a32nx/commit/3528d3e7d6c33e103f8d1b2c416452a0312c1731))

The beggining of the enroute segment is bogus as it goes "backwards" from the end of the SID... behaviour looks roughly correct here, the problem is with the route and nothing the FMGC can do will fix it.
![image](https://user-images.githubusercontent.com/9995998/162558844-7be3ee45-cef9-4ecb-89b3-c8324245085b.png)
![image](https://user-images.githubusercontent.com/9995998/162558854-5195904e-e748-45e0-a66f-808aabef8685.png)
Delete MOGTA, the offending fix, and we're okay again:
![image](https://user-images.githubusercontent.com/9995998/162558885-4585a2f9-c0a5-4596-bc55-9822a26aa0a0.png)

Onto the arrival... seems to just be a bad reversion due to distance + too big radius, combined with buggy transition logic. This may be fixed by #6850?
![image](https://user-images.githubusercontent.com/9995998/162558951-4fc0198b-7b23-4648-85ee-1787d13b02b2.png)
![image](https://user-images.githubusercontent.com/9995998/162558933-01adafc7-9234-4a7b-b3cd-394d0fa825a9.png)

Here we can see an obvious problem... as we don't have proper stringing logic (implemented in branch `cfms-20`) we end up backtracking, then going to GIBMA again:
![image](https://user-images.githubusercontent.com/9995998/162558970-e3ad132f-3b83-4f83-90cb-a205c4af291c.png)
![image](https://user-images.githubusercontent.com/9995998/162558980-91d31b91-7309-4b3a-ba26-2868912d5d5e.png)

Fix that up and we're all good:
![image](https://user-images.githubusercontent.com/9995998/162558994-fe3726a9-d7cb-4b39-8fff-a91fc3e0dd8c.png)
